### DISCUSSION TITLE
Too much less latency with others when connected in LAN to the server

### DISCUSSION THREAD

**Discussion by openaspace** [jamulussoftware/jamulus#3189] [Technical Support & Help] [answered]
Hello, 
when i connect to my local server via LAN at 2ms latency , if others users join in the room from the internet there is too much latency difference and i'm also in advance as ms... how i can solve?

**Comment by rdica** [comment]
Everyone that connects to your server will have higher latency simply because they aren't on your local network. You can't fix that except to have everyone come to your home and connect to your local network. That's the nature of networking in general.

**Reply by pljones** [reply]
And remember - everyone MUST only hear themselves through the server.  If you can hear yourself when you're not connected, or any of the other participants can hear when not connected, the need to reconfigure.

Otherwise hearing the local signal will prevent adapting to the latency.

**Reply by openaspace** [reply]
???

**Reply by pljones** [reply]
If you're hearing yourself locally, you won't play in time with others on the server.  If you can only hear yourself coming back from the server, you'll hear whether you're playing in time or not and adapt to play "sooner" to make up for the latency.

Each client's total latency will be the time from the triggering of the creation of the audio signal to the time you hear it in your ears.

That will depend on
- how they're creating the sound (variable)
- any signal processing applied before capture that introduces latency (variable)
- the time taken by the sound card to capture the audio input (variable)
- any signal processing applied after capture by the computer that introduces latency (variable)
- any signal processing applied by the computer between the hardware in and the Jamulus Client that introduces latency (variable)
- the buffering between the audio capture and the Jamulus Client (variable)
- the network delay between the Jamulus Client and the Jamulus Server outbound (variable)
- the buffering between the network and the Jamulus Server (variable)
- the time taken by the Jamulus Server to process the audio buffer (fixed - either 64 samples or 128 at 48K samples per second)
- the network delay between the Jamulus Server and the Jamulus Client inbound (variable)
- the buffering between the network and the Jamulus Client (variable)
- the buffering between the Jamulus Client and the audio output (variable)
- any signal processing applied by the computer between the Jamulus Client and the hardware out that introduces latency (variable)
- the time taken by the sound card to produced the audio output (variable)
- any signal processing applied after production that introduces latency (variable)
- how sound is monitored (variable)

All those "variable" amounts need to be kept as low as possible, of course.

**Comment by mcfnord** [comment]
Is there a cloud-based public server near you? It might give everyone similar delay.

**Reply by openaspace** [reply]
It's all ok thank you.
I can run on aws .... 
My question was only because I have 1gbps upload bandwidth at home

**Reply by mcfnord** [reply]
Sometimes this works great, but there are reasons why running a server on a home connection is a less reliable. Maybe you share some of your bandwidth with your neighbors, or your other devices. Every external participant has a signal coming to your house, and then another signal leaving your house. Any bottleneck in any direction will cause problems. But sometimes it does work great.

**Reply by openaspace** [reply, marked as answer]
Solved adding latency from my bheringer x32 mixer on the jamulus  aux send.

I don't want to pay for the server.
I already have a nice professional server in the office with 1gbps in out proxmox nvme 64gb ram , ping to Google 3 ms

**Reply by pljones** [reply]
You should be using a thru connection, not a send, to Jamulus.  Else you'll hear your local signal, rather than the yourself in the server mix.

**Reply by openaspace** [reply]
> You should be using a thru connection, not a send, to Jamulus.  Else you'll hear your local signal, rather than the yourself in the server mix.

I need to listen  myself at zero latency.. Everyone needs to listen to themselves without latency...
Not using the server latency

**Reply by jujudusud** [reply]
Please read the golden rule in the manual.

**Reply by openaspace** [reply]
> Please read the golden rule in the manuel.

Thank you but I'm not interested.. I mean..a musician needs to play in real time.. after what happens with other latency will be a fact..

A musician can't play listening itself with the latency..

It's a completely no sense of any investment in good audio hardware equipment.

I listen to myself using a big mixer with 16 channels with a top sound in real time.. have latency of my self it's like to don't play..

**Reply by jujudusud** [reply]
Do what ever you wanted...
If you don't respect the golden rule, dont ask anything here as it is a loss of time for developpers. ;-)

**Reply by mcfnord** [reply]
The golden rule is a bit overstated. It can be reasonable and useful to listen to myself via direct monitor rather than server stream, and is often recommended for percussionists at a distance.

**Reply by openaspace** [reply]
> The golden rule is a bit overstated. It can be reasonable and useful to listen to myself via direct monitor rather than server stream, and is often recommended for percussionists at a distance. 

No..I listen the mix of the digital mixer multichannel, not the bus that is submix mixed ad-hoc for others with limiter.. Compressor etc..

**Reply by jujudusud** [reply]
@mcfnord : overstated by whom?

**Reply by mcfnord** [reply]
It's literally called the Golden Rule, which is: _Do onto others as you would have them do onto you_. Right so far? So why do we in the North American West routinely recommend drummers violate it? Because they aren't causing anyone harm doing so, and breaking this rule lets us gain a drummer who might be quite a distance away.

The goal is to encourage people to hear where they appear in the mix, so they can develop a skill of playing ahead. But if your own signal is so distracting that it throws you off, it's sensible to put it aside. Hiphop rapping is another area where nothing is harmed by forgetting this recommendation. If you rap, it doesn't end up mattering if you rap precisely on the beat, or just operate in relation to the beat.

**Reply by jujudusud** [reply]
I think that's been made abundantly clear in the manual:
Before you connect to a Server, you should not be able to hear yourself. Listening to your sound from Jamulus (and not from yourself) is called “The Golden Rule” and enables you to play in time with others ([see the FAQ](https://jamulus.io/wiki/FAQ#why-should-i-not-listen-to-my-own-signal)).

**Reply by mcfnord** [reply]
Why is it called the Golden Rule? What it enables is learning play-ahead, which isn't always the goal, or realistic.

**Reply by openaspace** [reply]
Thank you to all.
We can close the discussion.
Thanks

**Reply by jujudusud** [reply]
> Why is it called the Golden Rule? What it enables is learning play-ahead, which isn't always the goal, or realistic.

I not the one who wrote this manual. Then I'm note the one to answer your question.

**Reply by openaspace** [reply]
My golden rule is 8000€ setup.
Bye

**Reply by jujudusud** [reply]
With additional investment, you can consider investing in the public network between you and the people you play with.
Bye.

**Comment by ann0see** [comment]
Closing and locking as opinion based. Golden rule is difficult to follow e.g. for acoustic musicians but can also help.


https://github.com/orgs/jamulussoftware/discussions/1580
